### DISCUSSION TITLE
Cool use of Plush nodes - optimize Flux prompts for T5-XXL and Clip-L text encoders!

### DISCUSSION THREAD

**Discussion by EnragedAntelope** [glibsonoran/Plush-for-ComfyUI#102] [Show and tell]
https://civitai.com/models/663699/t5-xxlclip-l-flux-prompt-optimizer-and-lora-autoloader-by-enragedantelope

**Comment by glibsonoran** [comment]
Nice workflow!

**Comment by an303042** [comment]
hey @EnragedAntelope and thanks for the workflow!
I've been using it and mostly enjoying it. 
I do have a question though.
I'm using groq, and for some reason, if my prompt is something a little loose like "come up with an idea for a photo of an item on the moon" and queue up some generations, it will go back to the same topics, or same ideas over and over again. It doesn't use the same prompt exactly - so it does go out and fetch a new prompt, but somehow groq will go back to the same 3-4 concepts over and over again.
Has anyone else experienced that?
Thanks!

**Reply by EnragedAntelope** [reply]
Glad you like! And I have the same issue. I don't know if part of the inference speedup is some sort of caching on the Groq side that I don't understand. But I have noticed that I can't just leave the same prompt in place and keep clicking "queue" or I will get mostly the same stuff. So I switch between their models with each queue, or slightly change the prompt/include a wildcard, or change the temperature, choose a different style, etc.

**Reply by an303042** [reply]
I see. Thanks for letting me know I'm not alone in this. Will try to think of easier ways to introduce some randomness.

**Reply by tazztone** [reply]
i think it is inherent of LLMs. that's why i use onebuttonprompt as user input for more randomness

**Reply by Jonseed** [reply]
I also use [One Button Prompt](https://github.com/AIrjen/OneButtonPrompt) to help inject randomness, diversity, and come up with prompt variants too. Very helpful nodes!

**Reply by tazztone** [reply]
if anyone comes up with something even more random than onebuttonprompt i would be glad to know.

**Reply by EnragedAntelope** [reply]
> if anyone comes up with something even more random than onebuttonprompt i would be glad to know.

I had LLMs make a ton of wildcards, then I have wildcards randomly be selected, then I tell LLMs to be creative (using system instruction + higher temperature) and enhance the prompt with additional styling/effects. 

Syntax for random wildcard selection in Impact Wildcard Processor in ComfyUI is {wildcard1|wildcard2|etc}. Example of a prompt I commonly use:

```
wearing __trendyoutfits/outfits__ with __enragedantelope/haircolor_merge__ __enragedantelope/hairstyle_merge__ hair, with a __enragedantelope/expression_merge__ expression, 

{__blink/dangerous_situation__|__enragedantelope/magical-realism-wildcards__|__enragedantelope/historical-action-wildcards__| __enragedantelope/humorous-situation-wildcards__|__enragedantelope/sci-fi-action-wildcards__|__enragedantelope/superhero-action-wildcards__|__enragedantelope/winter-prompts__|__enragedantelope/wholesome-scenes__|__enragedantelope/strange_creature_interactions__| __enragedantelope/dangerous-situation/dangerous_situation__ in a __enragedantelope/dangerous-situation/environment__, complicated by __enragedantelope/dangerous-situation/complicating_factor__|__enragedantelope/sports-moments__|__enragedantelope/job-prompts__|__enragedantelope/embarrassing-moment__|__enragedantelope/transportation-scenes__|__enragedantelope/fantasy-prompt__|__enragedantelope/fantasy-prompt2__|__enragedantelope/horror-prompt__}
```

**Reply by Jonseed** [reply]
Is the Impact Wildcard Processor like the Dynamic Prompts in Auto1111/Forge? Can it take wildcard files like Dynamic Prompts?

**Reply by EnragedAntelope** [reply]
> Is the Impact Wildcard Processor like the Dynamic Prompts in Auto1111/Forge? Can it take wildcard files like Dynamic Prompts?

Yes, exactly like that. Takes wildcards and takes the {|} syntax for random selection in-line. It's great :)

**Reply by Jonseed** [reply]
@EnragedAntelope check out this collection of "[billions of wildcards](https://civitai.com/models/138970/billions-of-wildcards-all-in-one)" in a single YAML file.

**Reply by tazztone** [reply]
i used those. but it it's just so many that it felt overwhelming to manually use them. what would be some smart way to use them ?

**Reply by Jonseed** [reply]
@tazztone in the [ImpactWildcardProcessor](https://github.com/ltdrdata/ComfyUI-Impact-Pack) node you can pick the "Select to Add Wildcard" field in the node, which will bring up a list of all the wildcards you have in your custom wildcards folder (which is obviously a very long list if you have this YAML file). But you can quickly filter this list with the open box at the top of the list. So type in "machine," for example, and it will filter the list of wildcards to only those that contain the word "machine," including some long wildcard prompts already put together under the "Bo" category at the top of the list. Click on the wildcard you want, and it will pop it into the prompt. This is a lot easier than using the wildcards manually.

**Reply by tazztone** [reply]
EDIT: only now i understood what the bo/ categories are for, they are like predefined recipes of multiple wildcards. 

thanks. tho i considered this to be "manually" as well.
however i just found you can basically prompt for the categories to get results from multiple wildcards at once by using * :
__properties/materials*/*__  
will get you every material

**Reply by EnragedAntelope** [reply]
> @EnragedAntelope check out this collection of "[billions of wildcards](https://civitai.com/models/138970/billions-of-wildcards-all-in-one)" in a single YAML file.

Thank you BTW. I am very familiar :) I try to keep up with wildcards from Civit and have a huge collection, and lately have really liked telling LLMs to make the wildcards for me based on ideas I have (like historical situations, horror scenes, etc.).

**Comment by EnragedAntelope** [comment]
FYI I just added a new version of this workflow that takes advantage of the additional parameters.

**Comment by glibsonoran** [comment]
Hey @EnragedAntelope,  A while back you had asked me if there was a way to load/use saved 'add_parameter' files (since my node allows you to save them as a file).  Well I figured out a way using the `Load Text` node from the `Custom Scripts` suite.  I made an example workflow called: 'How_to_retrieve_and_use_saved_parameter_files.png', it's in the Example_workflows subdirectory.  If you're interested.

**Comment by EnragedAntelope** [comment]
Thank you for the follow-up! I have since been keeping a long note in my workflows with the settings for copy/pasting, so this is a neater solution :) .

**Comment by glibsonoran** [comment]
Hey I added an Model-CLIP switch that allows multiple outputs from a single input.  It does rely on the nodes it's attached to be able to handle a 'None' value for when the switch is set to another path, but this has worked OK so far.  
It would be a way for you to switch between the Lora leg and the No-Lora leg on your workflow (see attached image).  Let me know if it works.

![Switch](https://github.com/user-attachments/assets/4e9b2efa-f682-4079-86ff-fc1b48cc0817)
